File: rag-engine/qingyu-rag/widget/example_token_endpoint.py
"""
qingyuweb.com 後端範例（Python + FastAPI）

跟 example-token-endpoint.js 做一樣的事，只是換成 Python，
如果 qingyuweb.com 的後端本來就是 Python 就直接照抄邏輯即可。

使用方式：
  export RAG_ENGINE_URL=https://api.your-domain.com
  export RAG_ENGINE_API_KEY=qyk_xxx...
  uvicorn example_token_endpoint:app --port 3000
"""
import os

import httpx
from fastapi import FastAPI, HTTPException

app = FastAPI()

RAG_ENGINE_URL = os.environ.get("RAG_ENGINE_URL", "http://localhost:8000")
RAG_ENGINE_API_KEY = os.environ.get("RAG_ENGINE_API_KEY")


@app.post("/api/widget-token")
async def get_widget_token():
    # 正式環境這裡通常還會先驗證使用者的 session / cookie，
    # 確認呼叫者真的有權限使用這個方塊，再決定要換哪個 tenant 的 token。
    if not RAG_ENGINE_API_KEY:
        raise HTTPException(500, "尚未設定 RAG_ENGINE_API_KEY")

    async with httpx.AsyncClient() as client:
        resp = await client.post(
            f"{RAG_ENGINE_URL}/auth/widget-token",
            headers={"Authorization": f"Bearer {RAG_ENGINE_API_KEY}"},
        )

    if resp.status_code != 200:
        raise HTTPException(502, "無法向 RAG Engine 取得 token")

    return resp.json()  # {token, expires_in_seconds}

File: apps/legacy/rag-engine/qingyu-rag/widget/example-token-endpoint.js
/**
 * qingyuweb.com 後端範例（Node.js + Express）
 *
 * 這段程式碼跑在 qingyuweb.com 自己的伺服器上，不是 RAG Engine 的一部分。
 * 目的：長期有效的 API Key 只存在這裡（環境變數），絕不送到瀏覽器；
 *       瀏覽器只拿得到短期、範圍受限的 widget JWT。
 *
 * 使用方式：
 *   RAG_ENGINE_URL=https://api.your-domain.com \
 *   RAG_ENGINE_API_KEY=qyk_xxx... \
 *   node example-token-endpoint.js
 */
const express = require("express");
const app = express();

const RAG_ENGINE_URL = process.env.RAG_ENGINE_URL || "http://localhost:8000";
const RAG_ENGINE_API_KEY = process.env.RAG_ENGINE_API_KEY;

if (!RAG_ENGINE_API_KEY) {
  console.warn("警告：尚未設定 RAG_ENGINE_API_KEY，/api/widget-token 會失敗");
}

app.post("/api/widget-token", async (req, res) => {
  // 這裡通常還會先驗證 req 裡的使用者 session / cookie，
  // 確認「這個瀏覽器真的屬於某個已登入、有權限使用這個方塊的使用者」，
  // 再決定要幫哪個 tenant_id 換 token（tenant_id 已經綁定在 API Key 上了）。

  try {
    const resp = await fetch(`${RAG_ENGINE_URL}/auth/widget-token`, {
      method: "POST",
      headers: { Authorization: `Bearer ${RAG_ENGINE_API_KEY}` },
    });

    if (!resp.ok) {
      return res.status(502).json({ error: "無法向 RAG Engine 取得 token" });
    }

    const data = await resp.json();
    res.json(data); // { token, expires_in_seconds }
  } catch (err) {
    console.error(err);
    res.status(500).json({ error: "內部錯誤" });
  }
});

app.listen(3000, () => console.log("qingyuweb.com 範例後端跑在 http://localhost:3000"));
